go/src/socialapi/workers/payment/webhook.go

```diff
@@ -0,0 +1,303 @@
+func GetHandler(name string) (StripeHandler, error) {
+	action, ok := stripeActions[name]
+	if ok {
+		return action, nil
+	}
+
+	return nil, errors.New("handler not found")
+}
+
+func formatCurrency(currencyStr string, amount uint64) string {
+	switch currencyStr {
+	case "USD", "usd":
+		currencyStr = "$"
+	default:
+		return ""
+	}
+
+	return fmt.Sprintf("%s%v", currencyStr, amount/100)
+}
+
+func chargeSucceededHandler(raw []byte) error {
+	var charge *stripe.Charge
+	err := json.Unmarshal(raw, &charge)
+	if err != nil {
+		return err
+	}
+
+	opts := getAmountOpts(charge)
+	eventName := "charge succeeded"
+
+	return sendEventForCustomer(charge.Customer.ID, eventName, opts)
+}
+
+func chargeFailedHandler(raw []byte) error {
+	var charge *stripe.Charge
+	err := json.Unmarshal(raw, &charge)
+	if err != nil {
+		return err
+	}
+
+	opts := getAmountOpts(charge)
+	eventName := "charge failed"
+
+	return sendEventForCustomer(charge.Customer.ID, eventName, opts)
+}
+
+func getAmountOpts(charge *stripe.Charge) map[string]interface{} {
+	amount := formatCurrency(string(charge.Currency), charge.Amount)
+	return map[string]interface{}{"amount": amount}
+}
+
+func customerSubscriptionCreatedHandler(raw []byte) error {
+	var req *stripe.Sub
+	err := json.Unmarshal(raw, &req)
+	if err != nil {
+		return err
+	}
+
+	eventName := fmt.Sprintf("subscribed to %s plan", req.Plan.ID)
+
+	return sendEventForCustomer(req.Customer.ID, eventName, nil)
+}
+
+func customerSubscriptionDeletedHandler(raw []byte) error {
+	var req *stripe.Sub
+	err := json.Unmarshal(raw, &req)
+	if err != nil {
+		return err
+	}
+
+	eventName := fmt.Sprintf("unsubscribed from %s plan", req.Plan.ID)
+
+	return sendEventForCustomer(req.Customer.ID, eventName, nil)
+}
+
+func customerSubscriptionUpdatedHandler(raw []byte) error {
+	var req *stripe.Sub
+	err := json.Unmarshal(raw, &req)
+	if err != nil {
+		return err
+	}
+
+	eventName := fmt.Sprintf("subscription of %s plan updated", req.Plan.ID)
+
+	return sendEventForCustomer(req.Customer.ID, eventName, nil)
+}
+
+func customerSubscriptionTrialWillEndHandler(raw []byte) error {
+	var req *stripe.Sub
+	err := json.Unmarshal(raw, &req)
+	if err != nil {
+		return err
+	}
+
+	eventName := fmt.Sprintf("trial of %s plan subscription will end", req.Plan.ID)
+
+	return sendEventForCustomer(req.Customer.ID, eventName, nil)
+}
+
+func invoiceCreatedHandler(raw []byte) error {
+	var invoice *stripe.Invoice
+	err := json.Unmarshal(raw, &invoice)
+	if err != nil {
+		return err
+	}
+
+	// we cant do anything to the closed invoices.
+	if invoice.Closed {
+		return nil
+	}
+
+	if invoice.Paid {
+		return nil
+	}
+
+	cus, err := customer.Get(invoice.Customer.ID, nil)
+	if err != nil {
+		return err
+	}
+
+	group, err := modelhelper.GetGroup(cus.Meta["groupName"])
+	// we might get events from other environments where we might not have the
+	// group in this env.
+	if err == mgo.ErrNotFound {
+		return nil
```

Comment:
since we have one testing account all the event will go the sandbox endpoint and it will log many unnecessary lines, if you think we could catch some issues with logging them, i could add "if config.Env == "sandbox" do not log etc, i didnt see a reason tho